eskipfile/remote.go

```diff
@@ -0,0 +1,146 @@
+package eskipfile
+
+import (
+	"errors"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/routing"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type remoteEskipFile struct {
+	preloaded       bool
+	remotePath      string
+	localPath       string
+	eskipFileClient *WatchClient
+	threshold       int
+	verbose         bool
+}
+
+// RemoteWatch creates a route configuration client with (remote) file watching. Watch doesn't follow file system nodes,
+// it always reads (or re-downloads) from the file identified by the initially provided file name.
+func RemoteWatch(rf string, th int, v bool) (routing.DataClient, error) {
```

Comment:
I think we should use here an Options structure instead of positional parameters, because it's fair to expect plenty more parameters in the future: timeout settings, tracing configuration, etc...

Reply:
Good idea. Solved!